### DISCUSSION TITLE
free-slip boundary implementation

### DISCUSSION THREAD

**Discussion by JCNNBY** [ProjectPhysX/FluidX3D#113] [General]
Dear Dr. Moritz Lehmann,

I have noticed the you apply free-slip boundary condition in your 'Concorde' demo in YouTube. Have you released such implementation? 

Many thanks for your reply.

Best

**Comment by ProjectPhysX** [comment]
Hi @JCNNBY,

the free-slip boundary implementation is still under construction.
Expect some delay, I'm quite busy with my new daytime job.

Kind regards,
Moritz

**Reply by JCNNBY** [reply]
Dear Dr. Moritz Lehmann,

I am thinking to contribute to your beautiful project by implementing the free-slip boundary condition. My first thought is to create a buffering zone separating the solid wall and the fluid. The DDFs are flipped according to the wall (symmetry plane) directions in the buffering zone so that the free-slip boundary condition can be fulfilled with 'Esoteric Pull'. Do you believe that will work?

Many thanks for your reply.

Best

**Reply by ProjectPhysX** [reply]
Hi @JCNNBY,

I already have a working free-slip implementation. The trick is to mirror-flip the loaded DDFs in-place with a large lookup-table of encoded boundary normal vectors. Writing that LUT manually was quite a pain.
Still there is 2 bigger issues:

1. Boundary force calculation needs to be different with mirrored DDFs and I'm not yet sure which variant is correct.
2. Revoxelized moving boundaries fail in a horrible way.

And then I need to check extension support and many more tests.
I haven't worked on this for the past 2 months (new job), but will resume soon.

I appreciate that you are so eager to contribute! But I can't accept Pull Requests for legal reasons. You may fork the repo and freely experiment with the code there, but please not merge back into the master branch.

Kind regards,
Moritz

**Reply by JCNNBY** [reply]
Dear Dr. Lehmann,

Much appreciate your detailed reply. I hope the following publication may be useful for your problems: 
https://hal.science/hal-02954802/document.

Best
